benchmarks/benchmark-fdot.C

```diff
@@ -0,0 +1,127 @@
+ * ========LICENCE========
+ * This file is part of the library FFLAS-FFPACK.
+ *
+ * FFLAS-FFPACK is free software: you can redistribute it and/or modify
+ * it under the terms of the  GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * ========LICENCE========
+ */
+
+#include "fflas-ffpack/fflas-ffpack-config.h"
+#include <iostream>
+#include <givaro/modular.h>
+
+#include "fflas-ffpack/fflas-ffpack.h"
+#include "fflas-ffpack/utils/timer.h"
+#include "fflas-ffpack/utils/fflas_io.h"
+#include "fflas-ffpack/paladin/parallel.h"
+#include "fflas-ffpack/paladin/fflas_plevel1.h"
+#include "fflas-ffpack/utils/args-parser.h"
+
+
+using namespace std;
+using namespace FFLAS;
+using namespace FFPACK;
+int main(int argc, char** argv) {
+
+	size_t iter      = 3;
+	size_t N      = 5000;
+ 	size_t BS      = 5000;
+//     size_t q = 101;
+	size_t p=0;
+	std::string file = "";
+  
+	Argument as[] = {
+		{ 'n', "-n N", "Set the dimension of the matrix C.",TYPE_INT , &N },
+// 		{ 'q', "-q Q", "Set the field characteristic (-1 for random).",         TYPE_INT , &q },
+		{ 'i', "-i R", "Set number of repetitions.",		TYPE_INT , &iter },
+		{ 's', "-i S", "Size of the integers.",				TYPE_INT , &BS },
+		{ 'p', "-p P", "0 for sequential, 1 for parallel.",	TYPE_INT , &p },
+		END_OF_ARGUMENTS
+	};
+  
+	FFLAS::parseArguments(argc,argv,as);
+  
+    typedef Givaro::ZRing<Givaro::Integer> Field;
+    Field F;
+    Givaro::GivRandom generator;
+    Givaro::IntegerDom IPD;
+
+// 	typedef Givaro::ModularBalanced<double> Field;
+// 	Field F(q);
+	
+	Field::Element_ptr A, B;
+	Field::Element d; F.init(d);
+  
+	Givaro::OMPTimer chrono, time; time.clear();
+  
+	for (size_t i=0;i<iter;++i){
+		A = fflas_new(F, N);
+		B = fflas_new(F, N);
+        PARFOR1D(j,N,SPLITTER(), {
+			IPD.random(generator,A[j],BS);
+			IPD.random(generator,B[j],BS);
+		});
+		
+// 		Field::RandIter G(F);
+// 		RandomMatrix (F, 1, N, A, lda, G);
+// 		RandomMatrix (F, 1, N, B, ldb, G);
+
+//         std::cerr << '['; for(size_t i=0; i<N; ++i) F.write(std::cerr,A[i]) << ' '; std::cerr << ']' << std::endl;
+//         std::cerr << '['; for(size_t i=0; i<N; ++i) F.write(std::cerr,B[i]) << ' '; std::cerr << ']' << std::endl;
+        
+        F.assign(d, F.zero);
+
+        size_t maxallowed_threads; // from env
+        PAR_BLOCK { maxallowed_threads=NUM_THREADS; }
```

Comment:
use a `-t` parameter defaulted to NUM_THREADS, as in `benchmark-fgemm.C` instead.